practice/case.py

```diff
@@ -0,0 +1,114 @@
+"""Case conversion between various cases
+
+Handles snake_case, CONSTANT_CASE, kebob-case, CamelCase
+Automatically detects case, assuming perfectly-typed
+
+Inputs:  word
+Output:  word converted to snake, camel, kebob and constant cases
+
+"""
+
+CASES = ['snake', 'camel', 'kebob', 'constant']
+
+
+def get_input_word():
+    """Get user word, requested to be changed."""
+    return input('What is your word? ')
+
+
+def detect_input_case(word):
+    """Detect input case based on word
+
+    >>> detect_input_case('snake_case')
+    'snake'
+    >>> detect_input_case('CamelCase')
+    'camel'
+    >>> detect_input_case('kebob-case')
+    'kebob'
+    >>> detect_input_case('CONSTANT_CASE')
+    'constant'
+    """
+    if '-' in word:
+        return 'kebob'
+    elif '_' in word:
+        if word[0].isupper():
+            return 'constant'
+        else:
+            return 'snake'
+    else:
+        return 'camel'
+
+
+def convert_to_snake(word, in_case):
+    """Convert word to snake case from input case.
+
+    >>> convert_to_snake('snake_case', 'snake')
+    'snake_case'
+    >>> convert_to_snake('CamelCase', 'camel')
+    'camel_case'
+    >>> convert_to_snake('kebob-case', 'kebob')
+    'kebob_case'
+    >>> convert_to_snake('CONSTANT_CASE', 'constant')
+    'constant_case'
+    """
+    out_word = []
+    if in_case == 'snake':
```

Comment:
Each one of these branches I think should be it's own function. Each is solving a totally separate and distinct problem and should be isolated so they can be run and tested independently.